lib/auth.js

```diff
@@ -297,7 +297,7 @@ const authClient = new Model({
         },
       };
 
-      return apiClient.put('/../users/password', data, config.jsonHeaders);
+      return makeCredentialHTTPRequest('PUT', '/../users/password', data, config.jsonHeaders);
```

Comment:
This will also need the API host domain + path URL, avoid defaulting to the window origin.